src/meta/src/backup_restore/backup.rs

```diff
@@ -0,0 +1,222 @@
+use risingwave_hummock_sdk::HummockSstableId;
+use tokio::task::JoinHandle;
+
+use crate::backup_restore::db_snapshot::{DbSnapshot, DbSnapshotBuilder};
+use crate::backup_restore::error::BackupError;
+use crate::backup_restore::storage::BackupStorageRef;
+use crate::backup_restore::DbSnapshotId;
+use crate::hummock::{HummockManagerRef, HummockVersionSafePoint};
+use crate::manager::{IdCategory, MetaSrvEnv};
+use crate::storage::MetaStore;
+use crate::MetaResult;
+
+#[derive(Clone)]
+pub enum BackupJobResult {
+    Cancelled,
+    Finished(DbSnapshot),
+    Failed,
+}
+
+/// `BackupJobHandle` tracks running job.
+struct BackupJobHandle {
+    job_id: u64,
+    join_handle: JoinHandle<BackupJobResult>,
+    hummock_version_safe_point: HummockVersionSafePoint,
+}
+
+impl BackupJobHandle {
+    pub fn new(
+        job_id: u64,
+        join_handle: JoinHandle<BackupJobResult>,
+        hummock_version_safe_point: HummockVersionSafePoint,
+    ) -> Self {
+        Self {
+            job_id,
+            join_handle,
+            hummock_version_safe_point,
+        }
+    }
+}
+
+pub type BackupManagerRef<S> = Arc<BackupManager<S>>;
+
+/// `BackupManager` manages lifecycle of all existent backups and the running backup job.
+pub struct BackupManager<S: MetaStore> {
+    env: MetaSrvEnv<S>,
+    hummock_manager: HummockManagerRef<S>,
+    backup_store: BackupStorageRef,
+    /// Existent database snapshot.
+    db_snapshots: parking_lot::RwLock<HashMap<DbSnapshotId, DbSnapshot>>,
+    /// Tracks the running backup job. Concurrent jobs is not supported.
+    running_backup_job: tokio::sync::Mutex<Option<BackupJobHandle>>,
+}
+
+impl<S: MetaStore> BackupManager<S> {
+    pub async fn new(
+        env: MetaSrvEnv<S>,
+        hummock_manager: HummockManagerRef<S>,
+        backup_store: BackupStorageRef,
+    ) -> MetaResult<Self> {
+        // TODO #6482 refine me: meta node actually only need SST manifest in memory.
+        let db_snapshots = backup_store
+            .list()
+            .await?
+            .into_iter()
+            .map(|s| (s.id, s))
+            .collect();
+        Ok(Self {
+            env,
+            hummock_manager,
+            backup_store,
+            db_snapshots: parking_lot::RwLock::new(db_snapshots),
+            running_backup_job: tokio::sync::Mutex::new(None),
+        })
+    }
+
+    #[cfg(test)]
+    pub fn for_test(env: MetaSrvEnv<S>, hummock_manager: HummockManagerRef<S>) -> Self {
+        Self {
+            env,
+            hummock_manager,
+            backup_store: Box::new(crate::backup_restore::DummyBackupStorage {}),
+            db_snapshots: Default::default(),
+            running_backup_job: Default::default(),
+        }
+    }
+
+    /// Starts a backup job in background. It's non-blocking.
+    /// Returns job id.
+    pub async fn start_backup_job(self: &Arc<Self>) -> MetaResult<u64> {
+        let mut guard = self.running_backup_job.lock().await;
```

Comment:
Shall we need async `Mutex` instead of an Atomic flag that indicates if there is a job running?

Reply:
`running_backup_job ` also holds a `HummockVersionSafePoint`, so I think using a `Mutex` keep code simpler (no need for multiple fields).